noobaa/noobaa-core
# Implement support for object prefix filtering in log-based replication optimization

### Explain the changes
1. Implement log-based optimization support for object filtering by user-provided prefix

### Issues: Fixed #xxx / Gap #xxx
1. https://bugzilla.redhat.com/show_bug.cgi?id=2281729

### Testing Instructions:
1. Create an OBC with a replication rule that utilizes `prefix` as well as `log_replication_info`
2. Write objects to the source bucket
3. Verify that only the objects that match the prefix were replicated
